src/modules/iotjs_module_pipe.c

```diff
@@ -176,29 +183,60 @@ static void iotjs_pipe_after_write(uv_write_t* req, int status) {
     iotjs_make_callback(callback, jerry_create_undefined(), &jargs);
     iotjs_jargs_destroy(&jargs);
   }
+  free(data);
   jerry_release_value(callback);
+  IOTJS_RELEASE(write_data);
   free(req);
 }
 
-JS_FUNCTION(WriteUtf8String) {
+static jerry_value_t doPipeWrite(iotjs_pipewrap_t_impl_t* _this,
+                                 const char* data, size_t size,
+                                 jerry_value_t callback) {
+  jerry_acquire_value(callback);
+  iotjs_pipe_write_data* write_data = IOTJS_ALLOC(iotjs_pipe_write_data);
+  write_data->data = malloc(sizeof(char) * size);
+  memcpy(write_data->data, data, size);
+  write_data->callback = callback;
+  uv_write_t* write_req = IOTJS_ALLOC(uv_write_t);
```

Comment:
How about make `iotjs_write_data` to be `iotjs_write_wrap` like the following:

```c
typedef struct iotjs_pipe_write_data {
  char* data;
  uv_write_t req;
  jerry_value_t callback;
} iotjs_pipe_write_data;
```

So we could only need to malloc once :)